packages/adapter-cloudflare/index.js

```diff
@@ -55,3 +66,43 @@ export default function () {
 		}
 	};
 }
+
+/**
+ * @param {import('../kit/types/internal').BuildData} build_data
+ * @returns {import('.').RoutesJSONSpec}
+ */
+function getRoutesJSONFromBuildData(build_data) {
+	return {
+		version: 1,
+		description: 'Generated by @sveltejs/adapter-cloudflare',
+		include: ['/*'],
+		exclude: [
+			`/${build_data.app_dir}/immutable/*`,
+			...build_data.manifest_data.assets
+				// We're being conservative by not excluding all assets in
+				// /static just yet. If there are any upstream auth rules to
+				// protect certain things (e.g. a PDF that requires auth),
+				// then we wouldn't want to prevent those requests from going
+				// to the user functions worker.
+				// We do want to show an example of a _routes.json that
+				// excludes more than just /_app/immutable/*, and favicons
+				// are a reasonable choice
+				.filter(({ file }) => file.includes('favicon'))
+				.map((asset) => asset.file)
```

Comment:
Can you elaborate on 'upstream auth rules'? Where would those be expressed? All static files are considered to be outside SvelteKit's purview (unless we added some platform-agnostic 'middleware' concept in future), so it feels like we could just list all static assets here

Reply:
> Where would those be expressed?

This would be the case of a Cloudflare Pages user adding logic to a catch-all functions route (e.g. `functions/_middleware.ts` or `functions/[[path.ts]]` or simply having a `_worker.js` file). They might be adding headers to static assets with their functions or protecting a certain files to only be accessible by emails ending in `@example.com`. Either way, we don't want to be _too prescriptive_ on whether or not functions runs before the asset or not.

Do note that _omitting a file from `excludes`_ will never accidentally break a user's function. The reverse is not true; A misplaced exclusion rule could prevent a function from being run that the user intended to be run. For that reason, we're intentionally being conservative on which assets are explicitly being excluded.

Further, there is a size limit for these descriptions. We're maxing out the _total number of rules_ - include or exclude - to 100. I'm sure plenty of sites would max out that rule limit with just excludes of static paths